### PR TITLE
Monster Zero and Piano+

### PR DESCRIPTION
**Pull request by Noztik**
tell victoresd to stop bugging me now

### CODE REVIEW

**Comment by swampservers** [comment]
there are a lot of conflicts with the piano in here. I'll move it to this repository

**Comment by Noztik** [comment]
should we replace the piano or just move it?
the synthesizer has the piano sounds in it anyway

**Comment by swampservers** [comment]
I added the piano files. Can you make the synthesizer use gmt_instrument_base instead of synthesizer_base, and get it to work? It looks like they mostly copy and pasted but i'm not sure

**Comment by Noztik** [comment]
why exactly do you want me to use gmt_instrument_base instead of just replacing it with synthesizer_base?

**Comment by swampservers** [comment]
my version of it is customized, to make seats work properly and fix a few other things

synthesizer_base looks like a copy and paste of the original gmt_instrument_base

**Comment by swampservers** [comment]
It looks like this just changes the functionality of the piano to act like a synthesizer, but still look like a piano. That's fine.

Can you remove the spurious materials: the original synthesizer texture and the spawnicons ( materials/vgui/entities/*)

**Comment by swampservers** [comment]
Looks good, added a few comments, will test later

**Comment by Noztik** [comment]
I only have one non-sip sound for the monster zero swep and I don't know what you mean by rebase but I think you forgot to switch the presents back to mystery crates and the present model was no longer in the workshop so it was just an error.

File: lua/weapons/weapon_monster/shared.lua [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+SWEP.Secondary.Ammo = "none"
+	
+SWEP.DrawAmmo = false
+SWEP.Spawnable = true
+
+SWEP.ViewModel	= "models/misc/monsterzero.mdl"
+SWEP.WorldModel	= "models/misc/monsterzero.mdl"
+
+function SWEP:Initialize()
+    self:SetWeaponHoldType("slam")
```
Comment (cannon) [inline lua/weapons/weapon_monster/shared.lua]: SetHoldType not SetWeaponHoldType

Comment (swampservers) [inline lua/weapons/weapon_monster/shared.lua]: SetHoldType not SetWeaponHoldType

File: lua/autorun/server/sv_monster.lua [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		net.Broadcast()
+	end
+	if ply.monsterCount >= 10 then
+		ply.cantStartmonster = true
+		ReleaseMonster(ply)
+	end
+end
+
+hook.Add("KeyRelease","DoMonsterHook",function(ply, key)
+	if key == IN_ATTACK and !ply.cantStartmonster and ply:GetActiveWeapon():GetClass() == "weapon_monster" then
```
Comment (swampservers) [inline lua/autorun/server/sv_monster.lua:23]: need to check IsValid(ply:GetActiveWeapon())

File: lua/autorun/server/sv_monster.lua [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		net.Broadcast()
+	end
+	if ply.cantStartmonster then
+		ply.fov = ply:GetFOV()
+		ply:SetWalkSpeed(300)
+		ply:SetRunSpeed(400)
+		ply:SetFOV(ply.fov+10,1)
+		timer.Simple(5,function()
+			ply.cantStartmonster=false
+			ply:SetWalkSpeed(200)
```
Comment (swampservers) [inline lua/autorun/server/sv_monster.lua]: Save the old walk and run speed, don't hard code. Also make sure that drinking twice doesn't give you permanent speed

**Comment by swampservers** [comment]
I'm pretty sure that the CurrentInstrument will not synchronize between clients. Can you change it to an NWInt? Don't set the NWInt on the client either.

File: lua/entities/gmt_instrument_base/init.lua [hunk 4]
```diff
@@ -17,11 +18,18 @@ function ENT:Initialize()
...
 		phys:Wake()
 	end
+	
+	self:SetNWInt("CurrentInstrument",1)
 
 	self:InitializeAfter()
 	
 end
 
+net.Receive("ChangeInstrument",function()
```
Comment (swampservers) [inline lua/entities/gmt_instrument_base/init.lua]: Not secure. check that the ent is valid, is a piano and that the player sending the message is sitting in the piano

**Pull request merged**